x/auth/vesting/types/vesting_account_test.go

```diff
@@ -926,6 +926,83 @@ func TestClawback(t *testing.T) {
 	require.Equal(t, int64(50), stakeAmt.Int64())
 }
 
+func TestClawbackBeforeStartTime(t *testing.T) {
+	c := sdk.NewCoins
+	fee := func(x int64) sdk.Coin { return sdk.NewInt64Coin(feeDenom, x) }
+	stake := func(x int64) sdk.Coin { return sdk.NewInt64Coin(stakeDenom, x) }
+	now := tmtime.Now()
+
+	// set up simapp and validators
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{}).WithBlockTime((now))
+	valAddr, val := createValidator(t, ctx, app, 100)
+	require.Equal(t, "stake", app.StakingKeeper.BondDenom(ctx))
+
+	lockupPeriods := []types.Period{
+		{Length: int64(12 * 3600), Amount: c(fee(1000), stake(100))},
+	}
+	vestingPeriods := []types.Period{
+		{Length: int64(8 * 3600), Amount: c(fee(200))},            // 8am
+		{Length: int64(1 * 3600), Amount: c(fee(200), stake(50))}, // 9am
+		{Length: int64(6 * 3600), Amount: c(fee(200), stake(50))}, // 3pm
+		{Length: int64(2 * 3600), Amount: c(fee(200))},            // 5pm
+		{Length: int64(1 * 3600), Amount: c(fee(200))},            // 6pm
+	}
+
+	bacc, origCoins := initBaseAccount()
+	_, _, funder := testdata.KeyTestPubAddr()
+	va := types.NewClawbackVestingAccount(bacc, funder, origCoins, now.Unix(), lockupPeriods, vestingPeriods)
+	addr := va.GetAddress()
+	app.AccountKeeper.SetAccount(ctx, va)
+
+	// fund the vesting account
+	err := simapp.FundAccount(app.BankKeeper, ctx, addr, c(fee(1000), stake(100)))
+	require.NoError(t, err)
+	require.Equal(t, int64(1000), app.BankKeeper.GetBalance(ctx, addr, feeDenom).Amount.Int64())
+	require.Equal(t, int64(100), app.BankKeeper.GetBalance(ctx, addr, stakeDenom).Amount.Int64())
+
+	// try delegating, clawback vesting account not allowed to delegate
+	_, err = app.StakingKeeper.Delegate(ctx, addr, sdk.NewInt(65), stakingtypes.Unbonded, val, true)
+	require.Error(t, err)
+
+	// undelegation should emit an error(delegator does not contain delegation)
+	_, err = app.StakingKeeper.Undelegate(ctx, addr, valAddr, sdk.NewDec(5))
+	require.Error(t, err)
+
+	// clawback prior to lockup time start, vesting time start
+	ctx = ctx.WithBlockTime(now.Add(7 * time.Hour))
+	_, _, dest := testdata.KeyTestPubAddr()
+	va2 := app.AccountKeeper.GetAccount(ctx, addr).(*types.ClawbackVestingAccount)
+	clawbackAction := types.NewClawbackAction(funder, dest, app.AccountKeeper, app.BankKeeper)
+	err = va2.Clawback(ctx, clawbackAction)
+	require.NoError(t, err)
+	app.AccountKeeper.SetAccount(ctx, va2)
+
+	// lockup period and vesting periods has been initialized
+	require.Equal(t, []types.Period{}, va2.LockupPeriods)
+	require.Equal(t, []types.Period{}, va2.VestingPeriods)
+
+	addrFeeBalance := app.BankKeeper.GetBalance(ctx, addr, feeDenom)
+	require.Equal(t, int64(1000), addrFeeBalance.Amount.Int64())
+	addrStakeBalance := app.BankKeeper.GetBalance(ctx, addr, stakeDenom)
+	require.Equal(t, int64(100), addrStakeBalance.Amount.Int64())
```

Comment:
```suggestion
	vestingAddrFeeBalance := app.BankKeeper.GetBalance(ctx, addr, feeDenom)
	require.Equal(t, int64(1000), vestingAddrFeeBalance.Amount.Int64())
	vestingAddrStakeBalance := app.BankKeeper.GetBalance(ctx, addr, stakeDenom)
	require.Equal(t, int64(100), vestingAddrStakeBalance.Amount.Int64())
```

Reply:
So these tokens didn't actually get clawedback? These are expected to all have gone to `dest`, right?

Reply:
I would expect them to either go back to `dest`, OR have the funder and the vesting account both not have access on the fund until the lockup period ended